system/include/emscripten/val.h

```diff
@@ -190,15 +191,14 @@ union GenericWireType {
     union {
         unsigned u;
         float f;
-        #if __ILP32__
-        const void* p;
-        #endif
+        // Use uint32_t for pointer values.  This limits us, for now, to 32-bit
+        // address ranges even on wasm64.  This is enforced by assertions below.
+        // TODO(sbc): Allow full 64-bit address range here under wasm64, most
```

Comment:
I assume if we updated this to 64-bit, we'd still want to use 53-bit ranges on the JS side though?

Reply:
Yes, I think they are separate concerns.